tools/ci/send-failure-message

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -e
+set -x
+
+if [[ "$ZULIP_BOT_KEY" != "" && "$GITHUB_REPOSITORY" == "zulip/zulip" ]]; then
+    BRANCH="$(python3 -c 'import sys; print(sys.argv[1].split("/")[-1])' "$GITHUB_REF")"
+    URI_ESCAPED_TOPIC="$(python3 -c 'import sys; import urllib.parse; print(urllib.parse.quote(sys.argv[1]))' "$BRANCH failing")"
+    curl -H "Content-Type: application/json" \
+        -X POST -i 'https://chat.zulip.org/api/v1/external/circleci?api_key='"$ZULIP_BOT_KEY"'&stream=automated%20testing&topic='"$URI_ESCAPED_TOPIC" \
```

Comment:
It's kinda weird that we're posting to the CircleCI webhook here; that at least deserves a comment that we're formatting our payload for that webhook?

Also would it make sense to write this in Python so that we can just create a dictionary and json.dump it for sending the failure message?

I think it's OK for that to be a follow-up commit.

Reply:
Okay will do!